### ISSUE TITLE
Allow retaining 0 uncertainties for tables

### ISSUE THREAD

**Issue by meliache** [josephwright/siunitx#639]
I have a table where I have values with uncertainties and a common exponent. Some table entries are 0 with 0 uncertainties, see the first screenshot below. I use `S[table-format=+4(1)e+1]` with separate uncertainties enabled via a global setting. With that setting, an entry like `0(0)` ended up being displayed as just `0` instead of `0 \pm 0`, so the uncertainty was dropped. In most cases this makes sense, if you don't have uncertainty don't show it, but in the table it seemed inconsistent.

I achieved the result in the screenshot via a hack: When I entered the 0-uncertainty as a floating point number, it wasn't dropped anymore, so I used `0(0.0)` for the zero entries. To avoid the decimal point from being printed, I then set `zero-decimal-to-integer=true`, which was needed to get the result shown here:

![image](https://user-images.githubusercontent.com/5121824/216640867-9ccd49e7-0642-404f-bc6e-2042dd0ad731.png)

I can't show the full table or the source code because it contains some not-yet-published sensitive data which I can't reveal yet. However, I also did a minimal working example below to make sure it works with a minimal setup:

``` latex
\documentclass{article}
\usepackage[separate-uncertainty=true]{siunitx}

\begin{document}
\begin{tabular}{lc}
  \hline
  source                                            & result                                     \\
  \hline
  \verb|\num{0(0)}|                                 & \num{0(0)}                                 \\
  \verb|\num{0(0.0)}|                               & \num{0(0.0)}                               \\
  \verb|\num[zero-decimal-to-integer=true]{0(0.0}}| & \num[zero-decimal-to-integer=true]{0(0.0)} \\
  \hline
\end{tabular}
\end{document}
```
![image](https://user-images.githubusercontent.com/5121824/216639721-b6585843-be25-4c8e-8c88-96f344575e4b.png)

I wouldn't necessarily call this a bug, maybe this is all intended, but I felt like there might be some unexpected behavior involved and wanted to report.

However, I would like for there to be an option to retain 0 uncertainties, like there's an option to retain 0 exponents.

**Comment by josephwright** [comment]
Which version do you have? The current release as `retain-zero-uncertainty` for this purpose.

**Comment by meliache** [comment]
According to `\listfiles` I locally have _siunitx 2022-04-04 v3.0.50_, which has the option and it seems to work:
![image](https://user-images.githubusercontent.com/5121824/216646505-822ff8d4-08fd-4e14-bf79-413d31dc8981.png)

I had read the documentation via `texdoc siunitx`, which for me opens `/usr/share/texmf-dist/doc/latex/siunitx/siunitx.pdf`, and weirdly, that manual has the date _2020/02/25_, which is why I couldn't find the option. I'm surprised, as I have installed texlive in  on archlinux via the package manager and assumed the documentation would match the version of installed packages.

Thanks, feel free to close the issue.

What I showed in the screenshots was compiled locally, but my colleagues uses overleaf (I push via git) and I know it has an older texlive release, hope the option is also available there.
